utils/src/types/instr.rs

```diff
@@ -0,0 +1,481 @@
+    U256Lt,
+    U256Le,
+    U256Gt,
+    U256Ge,
+    U256ModAdd,
+    U256ModSub,
+    U256ModMul,
+    U256BitAnd,
+    U256BitOr,
+    U256Xor,
+    U256SHL,
+    U256SHR,
+    I256ToU256,
+    I256ToByteVec,
+    U256ToI256,
+    U256ToByteVec,
+    ByteVecEq,
+    ByteVecNeq,
+    ByteVecSize,
+    ByteVecConcat,
+    AddressEq,
+    AddressNeq,
+    AddressToByteVec,
+    IsAssetAddress,
+    IsContractAddress,
+    Jump(I32),
+    IfTrue(I32),
+    IfFalse(I32),
+    Assert,
+    Blake2b,
+    Keccak256,
+    Sha256,
+    Sha3,
+    VerifyTxSignature,
+    VerifySecP256K1,
+    VerifyED25519,
+    NetworkId,
+    BlockTimeStamp,
+    BlockTarget,
+    TxId,
+    TxInputAddressAt,
+    TxInputsSize,
+    VerifyAbsoluteLocktime,
+    VerifyRelativeLocktime,
+    Log1,
+    Log2,
+    Log3,
+    Log4,
+    Log5,
+    ByteVecSlice,
+    ByteVecToAddress,
+    Encode,
+    Zeros,
+    U256To1Byte,
+    U256To2Byte,
+    U256To4Byte,
+    U256To8Byte,
+    U256To16Byte,
+    U256To32Byte,
+    U256From1Byte,
+    U256From2Byte,
+    U256From4Byte,
+    U256From8Byte,
+    U256From16Byte,
+    U256From32Byte,
+    EthEcRecover,
+    Log6,
+    Log7,
+    Log8,
+    Log9,
+    ContractIdToAddress,
+    LoadLocalByIndex,
+    StoreLocalByIndex,
+    Dup,
+    AssertWithErrorCode,
+    Swap,
+    BlockHash,
+    DEBUG(AVector<ByteString>),
+    TxGasPrice,
+    TxGasAmount,
+    TxGasFee,
+    I256Exp,
+    U256Exp,
+    U256ModExp,
+    VerifyBIP340Schnorr,
+    GetSegregatedSignature,
+    MulModN,
+    AddModN,
+    U256ToString,
+    I256ToString,
+    BoolToString,
+    GroupOfAddress,
+    LoadMutField(Byte),
+    StoreMutField(Byte),
+    ApproveAlph,
+    ApproveToken,
+    AlphRemaining,
+    TokenRemaining,
+    IsPaying,
+    TransferAlph,
+    TransferAlphFromSelf,
+    TransferAlphToSelf,
+    TransferToken,
+    TransferTokenFromSelf,
+    TransferTokenToSelf,
+    CreateContract,
+    CreateContractWithToken,
+    CopyCreateContract,
+    DestroySelf,
+    SelfContractId,
+    SelfAddress,
+    CallerContractId,
+    CallerAddress,
+    IsCalledFromTxScript,
+    CallerInitialStateHash,
+    CallerCodeHash,
+    ContractInitialStateHash,
+    ContractCodeHash,
+    MigrateSimple,
+    MigrateWithFields,
+    CopyCreateContractWithToken,
+    BurnToken,
+    LockApprovedAssets,
+    CreateSubContract,
+    CreateSubContractWithToken,
+    CopyCreateSubContract,
+    CopyCreateSubContractWithToken,
+    LoadMutFieldByIndex,
+    StoreMutFieldByIndex,
+    ContractExists,
+    CreateContractAndTransferToken,
+    CopyCreateContractAndTransferToken,
+    CreateSubContractAndTransferToken,
+    CopyCreateSubContractAndTransferToken,
+    NullContractAddress,
+    SubContractId,
+    SubContractIdOf,
+    ALPHTokenId,
+    LoadImmField(Byte),
+    LoadImmFieldByIndex,
+    PayGasFee,
+    MinimalContractDeposit,
+    CreateMapEntry(Byte, Byte),
+    MethodSelector(MethodSelector),
+    CallExternalBySelector(MethodSelector),
```

Comment:
[Nit] Not sure `MethodSelector` is not needed, perhaps we can use `MethodSelector(I32)` and `CallExternalBySelector(I32)` here.

Reply:
`MethodSelector` is equivalent to `i32`. The definition of `MethodSelector` is `struct MethodSelector(pub i32)`, which is similar to the type alias in scala.